env.go

```diff
@@ -207,19 +208,24 @@ type ReadinessProbe interface {
 	Ready(runnable Runnable) (err error)
 }
 
-// HTTPReadinessProbe checks readiness by making HTTP call and checking for expected HTTP status code.
+// HTTPReadinessProbe checks readiness by making HTTP or HTTPS call and checking for expected HTTP/HTTPS status code.
 type HTTPReadinessProbe struct {
 	portName                 string
 	path                     string
+	scheme                   string
 	expectedStatusRangeStart int
 	expectedStatusRangeEnd   int
 	expectedContent          []string
 }
 
-func NewHTTPReadinessProbe(portName string, path string, expectedStatusRangeStart, expectedStatusRangeEnd int, expectedContent ...string) *HTTPReadinessProbe {
+func NewHTTPReadinessProbe(portName, path, scheme string, expectedStatusRangeStart, expectedStatusRangeEnd int, expectedContent ...string) *HTTPReadinessProbe {
```

Comment:
I would prefer having `NewHTTPReadinessProbe` and `NewHTTPSReadinessProbe` so we don't need this extra `""` everywhere. WDYT?

we could have `newHTTPReadinessProve(portName, path, scheme. ..` we should share.

Reply:
I do not like the extra ""  everywhere. And the method should be backward-compatible. so I agree to have `NewHTTPReadinessProbe` and `NewHTTPSReadinessProbe`.

What do you mean `newHTTPReadinessProve(portName, path, scheme. ..`?